cmd/gateway/s3/gateway-s3.go

```diff
@@ -456,6 +458,13 @@ func (l *s3Objects) PutObject(ctx context.Context, bucket string, object string,
 	if err != nil {
 		return objInfo, minio.ErrorRespToObjectError(err, bucket, object)
 	}
+	tagStr, ok := opts.UserDefined[xhttp.AmzObjectTagging]
+	tagMap, err := getTagMap(tagStr)
+	if ok && err == nil && len(tagMap) > 0 {
+		if err = l.Client.PutObjectTagging(bucket, object, tagMap); err != nil {
+			return objInfo, minio.ErrorRespToObjectError(err, bucket, object)
+		}
+	}
```

Comment:
Instead of making another call for `PutObjectTagging`, isn't it better to directly add the tagging headers in `l.Client.PutObject`. This will save one server request.